generators/app/index.js

```diff
@@ -68,7 +73,9 @@ class NgxGenerator extends Generator {
         )})\n`
       );
       this.log(`${chalk.yellow("Make sure you don't have uncommitted changes before overwriting files!")}`);
-      this.insight.track('update', fromVersion, 'to', this.version);
+      if(!this.insight.optOut){
```

Comment:
no need for all this (and following checks) using if `insight.track` is noop'ed 😉